src/Program.cs

```diff
@@ -149,52 +153,44 @@ protected override async Task<bool> ProcessInternal()
                     throw new NotSupportedException($"Cannot convert command line argument --{kebabCasePropertyName} to {nameof(GeneratorSettingsTS)}.{propertyName} because type {propertyType} is not supported.");
                 }
             }
+        }
 
-            // process
-            var plugin = ExtensionsLoader.GetPlugin(
-                (await GetValue<bool?>("azure-arm") ?? false ? "Azure." : "") +
-                language +
-                (await GetValue<bool?>("fluent") ?? false ? ".Fluent" : "") +
-                (await GetValue<bool?>("testgen") ?? false ? ".TestGen" : ""));
-            Settings.PopulateSettings(plugin.Settings, Settings.Instance.CustomSettings);
-
-            void initializeSettings(CodeModelTS codeModel)
+        private async Task<IAnyPlugin> CreatePlugin()
+        {
+            bool azureArm = await GetValue<bool?>("azure-arm") ?? false;
+            IAnyPlugin plugin;
+            if (azureArm)
+            {
+                plugin = new PluginTSa();
+            }
+            else
             {
-                GeneratorSettingsTS generatorSettings = Singleton<GeneratorSettingsTS>.Instance;
-                codeModel.PackageName = Settings.Instance.PackageName;
-                codeModel.PackageVersion = Settings.Instance.PackageVersion;
-                codeModel.OutputFolder = generatorSettings.OutputFolder;
-                codeModel.ModelEnumAsUnion = generatorSettings.ModelEnumAsUnion;
-                codeModel.ModelDateTimeAsString = generatorSettings.ModelDateTimeAsString;
-                codeModel.GenerateMetadata = generatorSettings.GenerateMetadata;
-                codeModel.OptionalResponseHeaders = generatorSettings.OptionalResponseHeaders;
+                plugin = new PluginTS();
             }
 
-            using (plugin.Activate())
+            Settings.PopulateSettings(plugin.Settings, Settings.Instance.CustomSettings);
+
+            return plugin;
+        }
+
+        private async Task InitializeNamespace()
+        {
+            if (Settings.Instance.Namespace == null)
             {
-                Settings.Instance.Namespace = Settings.Instance.Namespace ?? CodeNamer.Instance.GetNamespaceName(altNamespace);
-                var codeModel = plugin.Serializer.Load(modelAsJson);
-                initializeSettings((CodeModelTS) codeModel);
-                codeModel = plugin.Transformer.TransformCodeModel(codeModel);
-                if (await GetValue<bool?>("sample-generation") ?? false)
-                {
-                    plugin.CodeGenerator.GenerateSamples(codeModel).GetAwaiter().GetResult();
-                }
-                else
-                {
-                    plugin.CodeGenerator.Generate(codeModel).GetAwaiter().GetResult();
-                }
+                string[] inputFiles = await GetValue<string[]>("input-file") ?? new[] { "" };

```

Comment:
why an array containing empty string instead of an empty array?

Reply:
I brought this change from the previous code, and I think they might've done it so that they could use `new[]` instead of `new string[0]`. I agree that it should be an empty array, though.